### PR TITLE
Allow for adding/removing root fields in the document editor

### PR DESCRIPTION
**Pull request by tjlav5**
Where applicable (in the AddDocument-step) allow for adding/removing root nodes from the document store.

image

### CODE REVIEW

File: src/components/Firestore/DocumentEditor/store.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ import * as actions from './actions';
 
 const reducer = createReducer<FirestoreMap, Action>({})
   .handleAction(actions.reset, (state, { payload }) => {
+    console.log('reset', payload);
```
Comment (tstirrat) [inline src/components/Firestore/DocumentEditor/store.tsx]: remove this

File: src/components/Firestore/DocumentEditor/index.tsx [hunk 2]
```diff
@@ -85,8 +95,10 @@ const DocumentEditor: React.FC<{
 const RootField: React.FC<{
   onChange?: (value: FirestoreMap) => void;
   areRootKeysMutable: boolean;
-}> = ({ onChange, areRootKeysMutable }) => {
+  limitToOneField: boolean;
```
Comment (tstirrat) [inline src/components/Firestore/DocumentEditor/index.tsx]: /** Some docs on exactly what this means*/

File: src/components/Firestore/DocumentEditor/store.tsx (hunk 1 above)
Comment (tjlav5) [reply src/components/Firestore/DocumentEditor/store.tsx]: Oops, done

File: src/components/Firestore/DocumentEditor/index.tsx (hunk 2 above)
Comment (tjlav5) [reply src/components/Firestore/DocumentEditor/index.tsx]: Done

**Pull request merged**